plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergResourceFactory.java

```diff
@@ -0,0 +1,100 @@
+import io.trino.spi.TrinoException;
+import io.trino.spi.classloader.ThreadContextClassLoader;
+import io.trino.spi.connector.ConnectorSession;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.SupportsNamespaces;
+import org.apache.iceberg.hive.HiveCatalog;
+import org.apache.iceberg.io.FileIO;
+
+import javax.inject.Inject;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+
+public class IcebergResourceFactory
+{
+    private final StaticMetastoreConfig metastoreConfig;
+    private final IcebergConfig icebergConfig;
+
+    @Inject
+    public IcebergResourceFactory(
+            StaticMetastoreConfig metastoreConfig,
+            IcebergConfig icebergConfig)
+    {
+        this.metastoreConfig = metastoreConfig;
+        this.icebergConfig = icebergConfig;
+    }
+
+    public Catalog getTableCatalog(ConnectorSession session)
+    {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+            Configuration configuration = getHadoopConfiguration();
+            Map<String, String> catalogProperties = getCatalogProperties(metastoreConfig, icebergConfig);
+            return CatalogUtil.loadCatalog(icebergConfig.getCatalogImpl(), icebergConfig.getCatalogName(), catalogProperties, configuration);
+        }
+    }
+
+    public SupportsNamespaces getSchemaCatalog(ConnectorSession session)
+    {
+        Catalog catalog = getTableCatalog(session);
+        if (catalog instanceof SupportsNamespaces) {
+            return (SupportsNamespaces) catalog;
+        }
+
+        throw new TrinoException(NOT_SUPPORTED, "catalog " + catalog.getClass().getName() + " does not support schema operations");
+    }
+
+    public FileIO getFileIO(ConnectorSession session)
+    {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+            Configuration configuration = getHadoopConfiguration();
+            Map<String, String> catalogProperties = getCatalogProperties(metastoreConfig, icebergConfig);
+            return CatalogUtil.loadFileIO(icebergConfig.getIoImpl(), catalogProperties, configuration);
+        }
+    }
+
+    private static Configuration getHadoopConfiguration()
+    {
+        try {
+            Configuration.addDefaultResource("hdfs-default.xml");
+            Configuration.addDefaultResource("hdfs-site.xml");
+            Configuration.addDefaultResource("hive-site.xml");
+            return new Configuration(true);
+        }
+        catch (RuntimeException e) {
+            return null;
+        }
+    }
+
+    private static Map<String, String> getCatalogProperties(StaticMetastoreConfig metastoreConfig, IcebergConfig icebergConfig)
```

Comment:
In our internal Iceberg connector, we have a well-defined set of catalog properties. However, in open source users should be able to input arbitrary catalog property. This can be done in Spark and Flink easily, but Trino is a bit harder due to its use of a POJO to parse and hold the config map. I can use a string config `additional-config` and convert it to a map, but that sounds like a hack. Any better suggestions?